lib/pair.ts

```diff
@@ -0,0 +1,266 @@
+import * as bitcoin from 'bitcoinjs-lib';
+import {encodeBtcAddress} from './encode';
+import {BigNumberish, BytesLike, BigNumber} from 'ethers';
+import {Obligation} from '../typechain/Obligation';
+import {Option} from '../typechain/Option';
+import {ITreasury} from '../typechain/ITreasury';
+import {OptionLib} from '../typechain/OptionLib';
+import {Ierc20} from '../typechain/Ierc20';
+import {BitcoinPriceFeed} from './feed';
+import {BtcAddress} from './bitcoin';
+
+interface EuropeanOption {
+  contract: string;
+  expiryTime: number;
+  windowSize: number;
+  strikePrice: BigNumber;
+  spotPrice: number;
```

Comment:
Do we want to have the spotPrice actually in here? Because that changes dynamically based on the BTC exchange rate and has little to do with the actual option contract.

One option I see is to declare a `EuropeanOption` with the following properties:
```
  contract: string;
  expiryTime: number;
  windowSize: number;
  strikePrice: BigNumber;
```
and a `RichEuropeanOption` that extends this with extra properties:
```
spotPrice: BigNumber // crrent BTC spot price
liquidity: BigNumber // current liquidity on the Uniswap pool
lastPrice: BigNumber // last traded price
```